orderapi/src/main/java/fi/hel/verkkokauppa/order/service/order/OrderService.java

```diff
@@ -109,11 +108,26 @@ public OrderAggregateDto getOrderWithItems(final String orderId) {
             List<OrderItem> items = this.orderItemService.findByOrderId(orderId);
             List<OrderItemMeta> metas = this.orderItemMetaService.findByOrderId(orderId);
             orderAggregateDto = orderTransformerUtils.transformToOrderAggregateDto(order, items, metas);
+
+            FlowStep flowStep = getFlowStepsByOrderId(orderId);
+            if (flowStep != null) {
+                orderAggregateDto.setFlowSteps(flowStepMapper.toDto(flowStep));
+            }
         }
 
         return orderAggregateDto;
     }
 
+    public FlowStep getFlowStepsByOrderId(String orderId) {
+        List<FlowStep> flowStep = flowStepRepository.findByOrderId(orderId);
+        if (flowStep.size() > 0) {
+            return flowStep.get(0);
```

Comment:
Could this be done with stream and filter? Example: https://github.com/City-of-Helsinki/verkkokauppa-core/blob/907aa7cadb73903e48acaf05b94249f9fcf5d0a1/merchantapi/src/main/java/fi/hel/verkkokauppa/configuration/service/MerchantService.java#L197 And https://github.com/City-of-Helsinki/verkkokauppa-core/blob/907aa7cadb73903e48acaf05b94249f9fcf5d0a1/merchantapi/src/main/java/fi/hel/verkkokauppa/configuration/service/MerchantService.java#L173

Reply:
Not here, because eventually the database will have quite a lot of flow steps, so then we would iterate through a huge list of elements to find the one with the right order id. I think the right way is to use query by field through the repository as it is done.